pypowsybl/dynamic.py

```diff
@@ -0,0 +1,190 @@
+
+
+class BranchSide(_Enum):
+    '''
+    warning the values are hardcoded in java layer
+    '''
+    ONE = "one"
+    TWO = "two"
+
+
+class ModelMapping:
+    def __init__(self) -> None:
+        self._handle = _pp.create_dynamic_model_mapping()
+
+    def add_alpha_beta_load(self, static_id: str, parameter_set_id: str) -> None:
+        self.add_all_dynamic_mappings(static_id=static_id,
+                                      parameter_set_id=parameter_set_id,
+                                      mapping_type=DynamicMappingType.ALPHA_BETA_LOAD)
+
+    def add_one_transformer_load(self, static_id: str, dynamic_param: str) -> None:
+        self.add_all_dynamic_mappings(static_id=static_id,
+                                      parameter_set_id=dynamic_param,
+                                      mapping_type=DynamicMappingType.ONE_TRANSFORMER_LOAD)
+
+    def add_omega_ref(self, generator_id: str) -> None:
+        self.add_all_dynamic_mappings(static_id=generator_id,
+                                      parameter_set_id="",
+                                      mapping_type=DynamicMappingType.OMEGA_REF)
+
+    def add_generator_synchronous_three_windings(self, static_id: str, dynamic_param: str) -> None:
+        self.add_all_dynamic_mappings(static_id=static_id,
+                                      parameter_set_id=dynamic_param,
+                                      mapping_type=DynamicMappingType.GENERATOR_SYNCHRONOUS_THREE_WINDINGS)
+
+    def add_generator_synchronous_three_windings_proportional_regulations(self, static_id: str, dynamic_param: str) -> None:
+        self.add_all_dynamic_mappings(static_id=static_id,
+                                      parameter_set_id=dynamic_param,
+                                      mapping_type=DynamicMappingType.GENERATOR_SYNCHRONOUS_THREE_WINDINGS_PROPORTIONAL_REGULATIONS)
+
+    def add_generator_synchronous_four_windings(self, static_id: str, dynamic_param: str) -> None:
+        self.add_all_dynamic_mappings(static_id=static_id,
+                                      parameter_set_id=dynamic_param,
+                                      mapping_type=DynamicMappingType.GENERATOR_SYNCHRONOUS_FOUR_WINDINGS)
+
+    def add_generator_synchronous_four_windings_proportional_regulations(self, static_id: str, dynamic_param: str) -> None:
+        self.add_all_dynamic_mappings(static_id=static_id,
+                                      parameter_set_id=dynamic_param,
+                                      mapping_type=DynamicMappingType.GENERATOR_SYNCHRONOUS_FOUR_WINDINGS_PROPORTIONAL_REGULATIONS)
+
+    def add_current_limit_automaton(self, static_id: str, dynamic_param: str, branch_side: BranchSide) -> None:
+        _pp.add_current_limit_automaton(
+            self._handle, static_id, dynamic_param, branch_side.value)
+
+    def add_all_dynamic_mappings(self, df: _pd.DataFrame = None, **kwargs: _Union[str, DynamicMappingType]) -> None:
```

Comment:
This is not exactly the design I had in mind: we should have the same approach as in the network class:
- all methods above should accept a dataframe or keywords as arguments (`add_current_limit_automaton` etc), to be able to add at once multiple instances of the model
- this method should be removed: we don't want to support mixing different models in the same dataframe, because as you underlined, they may have different arguments, hence different columns, which this approach forbids.